src/protocol/boolean/bitwise_lt.rs

```diff
@@ -0,0 +1,263 @@
+    ///   [h] = 0 0 0 1 0 0 0 0
+    /// ```
+    async fn step5(
+        g: &[Replicated<F>],
+        b: &[Replicated<F>],
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
+        record_id: RecordId,
+    ) -> Result<Vec<Replicated<F>>, BoxError> {
+        let mul = zip(repeat(ctx), zip(g, b))
+            .enumerate()
+            .map(|(i, (ctx, (&g_bit, &b_bit)))| {
+                let c = ctx.narrow(&BitOpStep::Step(i));
+                async move { c.multiply(record_id, g_bit, b_bit).await }
+            });
+        try_join_all(mul).await
+    }
+
+    /// Step 6. `[h] = Σ [h_i] where i=0..l-1`
+    ///
+    /// The interpretations is, `h` is 1 iff `a < b`
+    fn step6(h: &[Replicated<F>]) -> Replicated<F> {
+        h.iter()
+            .fold(Replicated::new(F::ZERO, F::ZERO), |acc, &x| acc + x)
+    }
+
+    #[allow(dead_code)]
+    pub async fn execute(
+        &self,
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
+        record_id: RecordId,
+    ) -> Result<Replicated<F>, BoxError> {
+        let mut e = Self::step1(self.a, self.b, ctx.narrow(&Step::AXorB), record_id).await?;
+        let f = Self::step2(&mut e, ctx.narrow(&Step::PrefixOr), record_id).await?;
+        let g = Self::step3_4(&f);
+        let h = Self::step5(&g, self.b, ctx.narrow(&Step::MaskLessThanBit), record_id).await?;
+        let result = Self::step6(&h);
+        Ok(result)
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+enum Step {
+    AXorB,
+    PrefixOr,
+    MaskLessThanBit,
+}
+
+impl crate::protocol::Substep for Step {}
+
+impl AsRef<str> for Step {
+    fn as_ref(&self) -> &str {
+        match self {
+            Self::AXorB => "a_xor_b",
+            Self::PrefixOr => "prefix_or",
+            Self::MaskLessThanBit => "mask_less_than_bit",
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::BitwiseLessThan;
+    use crate::{
+        error::BoxError,
+        ff::{Field, Fp2, Fp31},
+        protocol::{QueryId, RecordId},
+        secret_sharing::Replicated,
+        test_fixture::{make_contexts, make_world, share, validate_and_reconstruct, TestWorld},
+    };
+    use futures::future::try_join_all;
+    use rand::rngs::mock::StepRng;
+    use std::iter::zip;
+
+    type BitwiseShares = (
+        Vec<Replicated<Fp2>>,
+        (Vec<Replicated<Fp2>>, Vec<Replicated<Fp2>>),
+    );
+
+    fn fp31_to_bits(x: Fp31) -> Vec<Fp2> {
+        let x = u8::from(x);
+        (0..u8::BITS).map(|i| Fp2::from(x >> i & 1)).collect()
+    }
+
+    async fn bitwise_lt(a: Fp31, b: Fp31) -> Result<Fp2, BoxError> {
+        let world: TestWorld = make_world(QueryId);
+        let ctx = make_contexts::<Fp2>(&world);
+        let mut rand = StepRng::new(1, 1);
+
+        let a_bits = fp31_to_bits(a);
+        let b_bits = fp31_to_bits(b);
+
+        // Generate secret shares
+        #[allow(clippy::type_complexity)]
+        let ((a0, (a1, a2)), (b0, (b1, b2))): (BitwiseShares, BitwiseShares) = zip(a_bits, b_bits)
```

Comment:
This is still testing it in Fp2. Can you modify this to test the bits as shares of 0, 1 in a larger field?